src/dynsight/_internal/analysis/rdf.py

```diff
@@ -0,0 +1,155 @@
+
+            xyz_cols:
+                List of column names for XYZ format
+                files (e.g., `["Particle Type", "X", "Y", "Z"]`).
+        """
+        # Check number of args and setup the trajectory files
+        if topology_file is None:
+            self.trajectory = [Path(trajectory_file)]
+        else:
+            self.trajectory = [Path(topology_file), Path(trajectory_file)]
+
+        # Setup the pipeline
+        if topology_file is None:
+            self.pipeline = import_file(self.trajectory, columns=xyz_cols)
+        else:
+            self.pipeline = import_file(self.trajectory[0])
+            timeframes = LoadTrajectoryModifier()
+            timeframes.source.load(str(self.trajectory[1]))
+            self.pipeline.modifiers.append(timeframes)
+
+    def compute(
+        self,
+        cutoff: float,
+        bins: int = 200,
+        remove_atoms: str | None = None,
+        step: int = 1,
+    ) -> None:
+        """Compute the Radial Distribution Function (RDF).
+
+        Parameters:
+            cutoff:
+                Maximum distance for calculating the RDF.
+
+            bins:
+                Number of bins for the RDF calculation, Default is 200.
+
+            remove_atoms:
+                Selection OVITO expression for atoms
+                to be removed (e.g., 'ParticleType == 1').
+
+            step:
+                Frequency of time sampling for averaging
+                the RDF. Default is 1.
+
+        Important:
+            If you're using the `remove_atoms` argument to remove certain
+            particles from the RDF calculation, you need to use the
+            OVITO's expression syntax for selecting particles.
+
+            Example expressions:
+                - `"ParticleType == 1"`: Selects all particles of type 1.
+
+                - `"Position.X > 0"`: Selects particles with an x-coordinate
+                    greater than zero.
+
+                - `"ParticleIdentifier % 2 == 0"`: Selects particles with even
+                    particle identifiers.
+
+            Be sure to have this type of information stored in your
+            simulation file before using this command.
+
+            More information on OVITO's expression syntax can be found here:
+            https://www.ovito.org/manual/reference/pipelines/modifiers/expression_select.html
+        """
+        # Delete selected atoms (OVITO commands)
+        if remove_atoms is not None:
+            selection_modifier = ExpressionSelectionModifier(
+                expression=remove_atoms
+            )
+            self.pipeline.modifiers.append(selection_modifier)
+            self.pipeline.modifiers.append(DeleteSelectedModifier())
+
+        # Compute RDF
+        coord_modifier = CoordinationAnalysisModifier(
+            cutoff=cutoff, number_of_bins=bins
+        )
+        self.pipeline.modifiers.append(coord_modifier)
+
+        # Time averaging
+        averaging_modifier = TimeAveragingModifier(
```

Comment:
Make it clear in the documentation any parameters for the averaging?

Reply:
It should be an internal function that the user does not have the access to. The only think that matters is the frequency at which the user wants to average the RDF. I tried to explain this in a better way within the doc regarding the `frequency` parameter.